homeassistant/components/asuswrt/sensor.py

```diff
@@ -113,31 +138,31 @@ def __init__(
         self,
         coordinator: DataUpdateCoordinator,
         router: AsusWrtRouter,
-        sensor_type: str,
-        sensor_def: dict[str, Any],
+        description: AsusWrtSensorEntityDescription,
     ) -> None:
         """Initialize a AsusWrt sensor."""
         super().__init__(coordinator)
         self._router = router
-        self._sensor_type = sensor_type
-        self._attr_name = f"{DEFAULT_PREFIX} {sensor_def[SENSOR_NAME]}"
-        self._factor = sensor_def.get(SENSOR_FACTOR)
+        self.entity_description = description
+
+        self._sensor_key = description.key
+        self._factor = description.factor
+        self._precision = description.precision
```

Comment:
Why store anything that can already be derived from `self.entity_description` ?

Reply:
Fixed